test/v2/controllers/CreateUkPropertyPeriodSummaryControllerSpec.scala

```diff
@@ -0,0 +1,455 @@
+        Some(UkFhlPropertyIncome(
+          Some(5000.99),
+          Some(3123.21),
+          Some(UkPropertyIncomeRentARoom(
+            Some(532.12)
+          ))
+        )),
+        Some(UkFhlPropertyExpenses(
+          None,
+          None,
+          None,
+          None,
+          None,
+          None,
+          Some(988.18),
+          None,
+          None
+        ))
+      )),
+      Some(UkNonFhlProperty(
+        Some(UkNonFhlPropertyIncome(
+          Some(41.12),
+          Some(84.31),
+          Some(9884.93),
+          Some(842.99),
+          Some(31.44),
+          Some(UkPropertyIncomeRentARoom(
+            Some(947.66)
+          ))
+        )),
+        Some(UkNonFhlPropertyExpenses(
+          None,
+          None,
+          None,
+          None,
+          None,
+          None,
+          None,
+          None,
+          None,
+          None,
+          Some(988.18)
+        ))
+      ))
+    )
+
+  private val requestBodyJson = Json.parse(
+    """{
+      |    "fromDate": "2020-01-01",
+      |    "toDate": "2020-01-31",
+      |    "ukFhlProperty":{
+      |        "income": {
+      |            "periodAmount": 5000.99,
+      |            "taxDeducted": 3123.21,
+      |            "rentARoom": {
+      |                "rentsReceived": 532.12
+      |            }
+      |        },
+      |        "expenses": {
+      |            "premisesRunningCosts": 3123.21,
+      |            "repairsAndMaintenance": 928.42,
+      |            "financialCosts": 842.99,
+      |            "professionalFees": 8831.12,
+      |            "costOfServices": 484.12,
+      |            "other": 99282,
+      |            "travelCosts": 974.47,
+      |            "rentARoom": {
+      |                "amountClaimed": 8842.43
+      |            }
+      |        }
+      |    },
+      |    "ukNonFhlProperty": {
+      |        "income": {
+      |            "premiumsOfLeaseGrant": 42.12,
+      |            "reversePremiums": 84.31,
+      |            "periodAmount": 9884.93,
+      |            "taxDeducted": 842.99,
+      |            "otherIncome": 31.44,
+      |            "rentARoom": {
+      |                "rentsReceived": 947.66
+      |            }
+      |        },
+      |        "expenses": {
+      |            "premisesRunningCosts": 3123.21,
+      |            "repairsAndMaintenance": 928.42,
+      |            "financialCosts": 842.99,
+      |            "professionalFees": 8831.12,
+      |            "costOfServices": 484.12,
+      |            "other": 99282,
+      |            "residentialFinancialCost": 12.34,
+      |            "travelCosts": 974.47,
+      |            "residentialFinancialCostsCarriedForward": 12.34,
+      |            "rentARoom": {
+      |                "amountClaimed": 8842.43
+      |            }
+      |        }
+      |    }
+      |}
+      |""".stripMargin
+  )
+
+  private val requestBodyJsonConsolidatedExpense = Json.parse(
```

Comment:
Do we need separate consolidated/unconsolidated cases?

It doesn't matter what the JSON looks like anyway (this could be anything) as the RequestParser is mocked.

Reply:
I added bcoz foreign endpoint has both

Reply:
to be frank its not required, but bcoz we have 2 scenarios and it is implemented in foreign, I just added